tests/run/cpp_optional_temps.pyx

```diff
@@ -49,7 +49,8 @@ cdef extern from *:
 
 # this function just makes sure the output from the destructor can be captured by doctest
 cdef void print_C_destructor "print_C_destructor" () nogil:
-    print("~C()")
+    with gil:
```

Comment:
Can this use the `with gil` function declaration? (And I also wonder how this ever worked…)

Reply:
> And I also wonder how this ever worked…

I have a vague feeling that some simple print statements are able to work without the GIL (by design?). But it for some reason it stopped working, and it definitely isn't the point of the test so I didn't investigate further

Reply:
> I have a vague feeling that some simple print statements are able to work without the GIL (by design?)

Ah, sure. You switched the test from Py2 to Py3, which turned the `print` statement into a `print()` function call. Print statements implicitly acquire the GIL, for convenience. Print function calls are not so easy to spot. See the `InjectGILHandling` transform.